meal_mate/app/src/main/java/com/android/engineer/mealmate/view/features/home/HomeScreen.kt

```diff
@@ -21,21 +21,24 @@ import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.text.font.FontWeight
 import androidx.compose.ui.unit.dp
 import androidx.compose.ui.unit.sp
-import androidx.lifecycle.viewmodel.compose.viewModel
+import androidx.hilt.navigation.compose.hiltViewModel
 import androidx.navigation.NavHostController
 import com.android.engineer.mealmate.R
+import com.android.engineer.mealmate.data.utils.DURATION
+import com.android.engineer.mealmate.data.utils.FRUITS
+import com.android.engineer.mealmate.data.utils.SERVES
 import com.android.engineer.mealmate.ui.theme.OrangeOnPrimary
 import com.android.engineer.mealmate.ui.theme.OrangePrimary
 import com.android.engineer.mealmate.data.utils.STATIC_BREAK_FAST_IMAGE
-import com.android.engineer.mealmate.view.utils.custom_views.MealIconButton
 import com.android.engineer.mealmate.view.utils.custom_views.MealImageLoading
+import com.android.engineer.mealmate.view.utils.custom_views.MealLottieAnimation
 import com.android.engineer.mealmate.view.utils.custom_views.MealSearchView
 import com.android.engineer.mealmate.view.utils.custom_views.MealText
 
 @Composable
-fun HomeScreen(navHostController: NavHostController, userName: String, paddingValues: PaddingValues) {
-    val viewModel = viewModel<RecipeViewModel>()
-
+fun HomeScreen(navHostController: NavHostController, paddingValues: PaddingValues) {
+    val viewModel = hiltViewModel<RecipeViewModel>()
```

Comment:
i think you can directly pass this as an function param
 ``` suggestion
 HomeScreen(navHostController: NavHostController, paddingValues: PaddingValues, viewModel: RecipeViewModel = hiltViewModel())
 ```

Reply:
@nasreekar Why any issues with this approach?

Reply:
no issues. just another suggestion. both works :)